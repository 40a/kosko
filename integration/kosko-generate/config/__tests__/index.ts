import execa from "execa";
import { dirname } from "path";
import { runCLI } from "../../../run";

let args: string[];
let result: execa.ExecaReturns;

beforeEach(async () => {
  result = await runCLI(args, {
    cwd: dirname(__dirname)
  });
});

describe("given config file", () => {
  beforeAll(() => {
    args = ["generate"];
  });

  test("should return status code 0", () => {
    expect(result.code).toEqual(0);
  });

  test("should apply configs", () => {
    expect(result.stdout).toMatchSnapshot();
  });
});
