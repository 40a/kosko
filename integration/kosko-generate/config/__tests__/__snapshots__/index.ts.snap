// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given config file should apply configs 1`] = `
"apiVersion: v1
kind: ConfigMap
metadata:
  name: a"
`;
